topology/intel/ssp-nhlt.c

```diff
@@ -0,0 +1,860 @@
+
+#define SOF_DAI_FMT_NB_NF       (0 << 8) /**< normal bit clock + frame */
+#define SOF_DAI_FMT_NB_IF       (2 << 8) /**< normal BCLK + inv FRM */
+#define SOF_DAI_FMT_IB_NF       (3 << 8) /**< invert BCLK + nor FRM */
+#define SOF_DAI_FMT_IB_IF       (4 << 8) /**< invert BCLK + FRM */
+
+#define SOF_DAI_FMT_CBP_CFP     (0 << 12) /**< codec bclk provider & frame provider */
+#define SOF_DAI_FMT_CBC_CFP     (2 << 12) /**< codec bclk consumer & frame provider */
+#define SOF_DAI_FMT_CBP_CFC     (3 << 12) /**< codec bclk provider & frame consumer */
+#define SOF_DAI_FMT_CBC_CFC     (4 << 12) /**< codec bclk consumer & frame consumer */
+
+#define SOF_DAI_FMT_FORMAT_MASK         0x000f
+#define SOF_DAI_FMT_CLOCK_MASK          0x00f0
+#define SOF_DAI_FMT_INV_MASK            0x0f00
+#define SOF_DAI_FMT_CLOCK_PROVIDER_MASK 0xf000
+
+/* SSCR0 bits */
+#define SSCR0_DSIZE(x)	SET_BITS(3, 0, (x) - 1)
+#define SSCR0_FRF	MASK(5, 4)
+#define SSCR0_MOT	SET_BITS(5, 4, 0)
+#define SSCR0_TI	SET_BITS(5, 4, 1)
+#define SSCR0_NAT	SET_BITS(5, 4, 2)
+#define SSCR0_PSP	SET_BITS(5, 4, 3)
+#define SSCR0_ECS	BIT(6)
+#define SSCR0_SSE	BIT(7)
+#define SSCR0_SCR_MASK	MASK(19, 8)
+#define SSCR0_SCR(x)	SET_BITS(19, 8, x)
+#define SSCR0_EDSS	BIT(20)
+#define SSCR0_NCS	BIT(21)
+#define SSCR0_RIM	BIT(22)
+#define SSCR0_TIM	BIT(23)
+#define SSCR0_FRDC(x)	SET_BITS(26, 24, (x) - 1)
+#define SSCR0_ACS	BIT(30)
+#define SSCR0_MOD	BIT(31)
+
+/* SSCR1 bits */
+#define SSCR1_RIE	BIT(0)
+#define SSCR1_TIE	BIT(1)
+#define SSCR1_LBM	BIT(2)
+#define SSCR1_SPO	BIT(3)
+#define SSCR1_SPH	BIT(4)
+#define SSCR1_MWDS	BIT(5)
+#define SSCR1_TFT_MASK	MASK(9, 6)
+#define SSCR1_TFT(x)	SET_BITS(9, 6, (x) - 1)
+#define SSCR1_RFT_MASK	MASK(13, 10)
+#define SSCR1_RFT(x)	SET_BITS(13, 10, (x) - 1)
+#define SSCR1_EFWR	BIT(14)
+#define SSCR1_STRF	BIT(15)
+#define SSCR1_IFS	BIT(16)
+#define SSCR1_PINTE	BIT(18)
+#define SSCR1_TINTE	BIT(19)
+#define SSCR1_RSRE	BIT(20)
+#define SSCR1_TSRE	BIT(21)
+#define SSCR1_TRAIL	BIT(22)
+#define SSCR1_RWOT	BIT(23)
+#define SSCR1_SFRMDIR	BIT(24)
+#define SSCR1_SCLKDIR	BIT(25)
+#define SSCR1_ECRB	BIT(26)
+#define SSCR1_ECRA	BIT(27)
+#define SSCR1_SCFR	BIT(28)
+#define SSCR1_EBCEI	BIT(29)
+#define SSCR1_TTE	BIT(30)
+#define SSCR1_TTELP	BIT(31)
+
+#if CONFIG_BAYTRAIL || CONFIG_CHERRYTRAIL
+/* SSCR2 bits */
+#define SSCR2_URUN_FIX0	BIT(0)
+#define SSCR2_URUN_FIX1	BIT(1)
+#define SSCR2_SLV_EXT_CLK_RUN_EN	BIT(2)
+#define SSCR2_CLK_DEL_EN		BIT(3)
+#define SSCR2_UNDRN_FIX_EN		BIT(6)
+#define SSCR2_FIFO_EMPTY_FIX_EN		BIT(7)
+#define SSCR2_ASRC_CNTR_EN		BIT(8)
+#define SSCR2_ASRC_CNTR_CLR		BIT(9)
+#define SSCR2_ASRC_FRM_CNRT_EN		BIT(10)
+#define SSCR2_ASRC_INTR_MASK		BIT(11)
+#elif CONFIG_CAVS || CONFIG_HASWELL || CONFIG_BROADWELL
+#define SSCR2_TURM1		BIT(1)
+#define SSCR2_PSPSRWFDFD	BIT(3)
+#define SSCR2_PSPSTWFDFD	BIT(4)
+#define SSCR2_SDFD		BIT(14)
+#define SSCR2_SDPM		BIT(16)
+#define SSCR2_LJDFD		BIT(17)
+#define SSCR2_MMRATF		BIT(18)
+#define SSCR2_SMTATF		BIT(19)
+#endif
```

Comment:
NHLT only supports CAVS, suggest removing all irrelevant references to older platforms, it's dead code